src/stdlib_array.f90

```diff
@@ -0,0 +1,60 @@
+! SPDX-Identifier: MIT
+
+!> Module for index manipulation and general array handling
+module stdlib_array
+  implicit none
+  private
+
+  public :: trueloc, falseloc
+
+contains
+
+  !> Return the positions of the true elements in array
+  pure function trueloc(array, lbound) result(loc)
+    !> Mask of logicals
+    logical, intent(in) :: array(:)
+    !> Lower bound of array to index
+    integer, intent(in), optional :: lbound
+    !> Locations of true elements
+    integer :: loc(count(array))
```

Comment:
I wonder how it will perform if `size(array) == 0`? I suspect it will still work (and return a logical array of length 0), but it would be good to check this and either include such a test in the test-suite, or else document that it won't work.

Reply:
I won't repeat this comment below, but obviously it applies to all the routines